envoy/cilium/npds.proto

```diff
@@ -115,3 +120,36 @@ message HttpNetworkPolicyRule {
   // Optional. If empty, matches any HTTP request.
   repeated envoy.api.v2.route.HeaderMatcher headers = 1;
 }
+
+// A set of network policy rules that match Kafka requests.
+message KafkaNetworkPolicyRules {
+  // The set of Kafka network policy rules.
+  // A Kafka request is matched if any of its rules matches the request.
+  // Required and may not be empty.
+  repeated KafkaNetworkPolicyRule kafka_rules = 1 [(validate.rules).repeated.min_items = 1];
+}
+
+// A Kafka network policy rule, as a conjunction of predicates on Kafka requests.
+// If all the predicates of a rule match a Kafka request, the request is allowed. Otherwise, it is
+// denied.
+message KafkaNetworkPolicyRule {
```

Comment:
Should we also include Kafka `Roles` or that will already be resolved to low level apiKeys when we check these values?

Reply:
The roles will be resolved when converting into those rules.